app/controllers/api_new/users_controller.rb

```diff
@@ -12,4 +12,14 @@ class ApiNew::UsersController < ApiNew::ApiController
 	def current
 		@user = current_user
 	end
+
+	# get /api_new/users/current_nonprofit/object_events
+	def current_nonprofit_object_events
+		np_houid = current_user.roles.where(host_type: 'Nonprofit').first&.host&.houid
+		if np_houid
+			redirect_to api_new_nonprofit_object_events_path(np_houid)
```

Comment:
Any get query params need to be passed. The route helper has the ability accept those params as arguments if you pass them to it.

Additionally, you need to test that provided params are correctly passed.